lib/bolt/inventory/inventory2.rb

```diff
@@ -17,12 +17,19 @@ def initialize(data, config = nil, plugins: nil, target_vars: {}, target_facts:
         # Config is saved to add config options to targets
         @config = config || Bolt::Config.default
         @data = data ||= {}
-        @groups = Group2.new(data.merge('name' => 'all'))
+        @data = Bolt::Util.walk_vals(data) do |value|
+          if value.is_a?(Hash) && value.include?('_plugin')
+            # How do we reliably instantiate new plugin object by name?
+            Object.const_get("Bolt::Plugin::#{value['_plugin'].capitalize}").new.inventory_config_lookup(value)
+          else
+            value
+          end
+        end
+        @groups = Group2.new(@data.merge('name' => 'all'))
```

Comment:
In this case the init method has a parameter called `data` and an instance variable `@data`. When I iterate over the instance variable and convert `{'_plugin' => promt, ...}` I want to instantiate `@groups` with the modified `@data` version instead of `data` version. Do we intentionally *not* want to initialize `Group2` with the instance variable version?